Introduce BP_PHP_NGINX_ENABLE_HTTPS to enable https

arjun024 opened the pull request:
`ssl` directive has been deprecated since 1.15.0,
and is removed in nginx 1.25.1.
This is replaced by the configuration option to add the value `ssl` to the `listen` directive.

In the pre-deprecation world, a user can add a user-defined conf file with the app to turn on ssl. In the new world, since the listen directive is provided as part of the core nginx conf generated by this buildpack, users are left with no option (you can't override a directive).

This change solves the issue by allowing users to set `BP_PHP_NGINX_ENABLE_HTTPS=true`

Fixes https://github.com/paketo-buildpacks/php-web/issues/139
Fixes https://github.com/paketo-buildpacks/php-nginx/issues/21
Fixes failure of https://github.com/paketo-buildpacks/php/pull/724 with env var addition in tests.

Links:
http://hg.nginx.org/nginx/rev/0aaa09927703
http://nginx.org/en/CHANGES
http://nginx.org/en/docs/http/ngx_http_core_module.html#listen

## Checklist
<!-- Please confirm the following -->
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

sophiewigmore reviewed (dismissed):
thanks for putting this change in! ⭐  we should document this change on the website docs as well

arjun024 commented:
@sophiewigmore Thanks for looking into this.
1. Fixed integration tests
2. Added docs to website https://github.com/paketo-buildpacks/paketo-website/pull/696
